src/dymaptic.GeoBlazor.Core/Scripts/arcGisJsInterop.ts

```diff
@@ -737,9 +738,21 @@ export async function addWidget(widget: any, viewId: string): Promise<void> {
                 }
                 break;
             case 'legend':
-                newWidget = new Legend({
+                const legend = new Legend({
                     view: view
                 });
+                newWidget = legend;
+
+                break;
+            case 'home':
+                const homeBtn = new Home({
+                    view: view,
+                });
+                newWidget = homeBtn;
+                if (widget.home !== undefined && widget.home !== null) {
```

Comment:
752-754 should be checking `widget.label` and then setting `homeBtn.label = widget.label`.
Then do the same pattern for `widget.iconClass` (check for undefined/null, set on `homeBtn`.
You do _not_ need to set `view.ui.add` here, it happens after the switch statement.

Reply:
adjusted and updated